File: Examples/Python/tcp_connection.py
import connection
import helpers
import ximu3

if helpers.yes_or_no("Search for connections?"):
    network_announcement = ximu3.NetworkAnnouncement()

    result = network_announcement.get_result()

    if network_announcement.get_result() != ximu3.RESULT_OK:
        raise Exception(f"Network announcement failed. {ximu3.result_to_string(result)}.")

    messages = network_announcement.get_messages_after_short_delay()

    if not messages:
        raise Exception("No TCP connections available")

    print(f"Found {messages[0].device_name} {messages[0].serial_number}")

    connection_info = messages[0].to_tcp_connection_info()

    connection.run(connection_info)
else:
    connection_info = ximu3.TcpConnectionInfo("192.168.1.1", 7000)  # replace with actual connection info

    connection.run(connection_info)

File: Examples/CSharp/x-IMU3-Examples/Examples/TcpConnection.cs
namespace Ximu3Examples
{
    class TcpConnection : Connection
    {
        public TcpConnection()
        {
            if (Helpers.YesOrNo("Search for connections?"))
            {
                Ximu3.NetworkAnnouncement networkAnnouncement = new();

                Ximu3.CApi.XIMU3_Result result = networkAnnouncement.GetResult();

                if (result != Ximu3.CApi.XIMU3_Result.XIMU3_ResultOk)
                {
                    Console.WriteLine("Network announcement failed. " + Ximu3.Helpers.ToString(Ximu3.CApi.XIMU3_result_to_string(result)) + ".");
                    return;
                }

                Ximu3.CApi.XIMU3_NetworkAnnouncementMessage[] messages = networkAnnouncement.GetMessagesAfterShortDelay();

                if (messages.Length == 0)
                {
                    Console.WriteLine("No TCP connections available");
                    return;
                }

                Console.WriteLine("Found " + Ximu3.Helpers.ToString(messages[0].device_name) + " " + Ximu3.Helpers.ToString(messages[0].serial_number));

                Ximu3.TcpConnectionConfig config = new(messages[0]);

                Run(config);
            }
            else
            {
                Ximu3.TcpConnectionConfig config = new("192.168.1.1", 7000); // replace with actual connection config

                Run(config);
            }
        }
    }
}
